dsl/camel-jbang/camel-jbang-plugin-k/src/main/java/org/apache/camel/dsl/jbang/core/commands/k/Agent.java

```diff
@@ -0,0 +1,424 @@
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+@CommandLine.Command(name = Agent.ID, description = "TODO", sortOptions = false)
+public class Agent extends CamelCommand {
+    private static final Logger LOGGER = LoggerFactory.getLogger(Agent.class);
+
+    private static final ObjectMapper MAPPER = new ObjectMapper()
+            .setSerializationInclusion(JsonInclude.Include.NON_EMPTY);
+
+    private static final Map<String, BiConsumer<CamelCatalog, SourceMetadata>> COMPONENT_CUSTOMIZERS;
+
+    public static final String ID = "agent";
+    public static final String STUB_PATTERN = "*";
+    public static final String MIME_TYPE_JSON = "application/json";
+    public static final String CONTENT_TYPE_HEADER = "Content-Type";
+
+    static {
+        // the core CircuitBreaker reifier fails as it depends on a specific implementation provided by
+        // a dedicated component/module, but as we are orly inspecting the route to determine its capabilities,
+        // we can safely disable the EIP with a disabled/stub reifier.
+        ProcessorReifier.registerReifier(CircuitBreakerDefinition.class, DisabledReifier::new);
+
+        COMPONENT_CUSTOMIZERS = new HashMap<>();
+        COMPONENT_CUSTOMIZERS.put("platform-http", (catalog, meta) -> {
+            meta.capabilities.put(
+                    Capability.PlatformHttp,
+                    catalog
+                            .findCapabilityRef(Capability.PlatformHttp.getValue())
+                            .orElseGet(() -> new EntityRef(null, null)));
+        });
+
+    }
+
+    @CommandLine.Option(
+                        names = { "--listen-host" },
+                        description = "")
+    String host = "localhost";
+
+    @CommandLine.Option(
+                        names = { "--listen-port" },
+                        description = "")
+    int port = 8081;
+
+    @CommandLine.Option(
+                        names = { "--runtime-version" },
+                        description = "To use a different runtime version than the default version")
+    String runtimeVersion;
+
+    @CommandLine.Option(
+                        names = { "--runtime" },
+                        converter = RuntimeTypeConverter.class,
+                        description = "Runtime (spring-boot, quarkus, camel-main)")
+    RuntimeType runtimeType = RuntimeType.camelMain;
+
+    @CommandLine.Option(
+                        names = { "--repos" },
+                        description = "Comma separated list of additional maven repositories")
+    String repos;
+
+    public Agent(CamelJBangMain main) {
+        super(main);
+    }
+
+    @Override
+    public Integer doCall() throws Exception {
+        Vertx vertx = null;
+        HttpServer server = null;
+
+        try {
+            CountDownLatch latch = new CountDownLatch(1);
+
+            vertx = Vertx.vertx();
+            server = serve(vertx).toCompletableFuture().get();
+
+            latch.await();
```

Comment:
What triggers this latch to complete, it seems its a local latch only

Reply:
yeah, it is to keep the server up as in fact the servers does not block

Reply:
okay maybe add a code comment about that, and if the server is asked to terminate via SIGTERM then it will just hang wont it? You may need to do like camel-main and have a jvm shutdown hook, but that can be done in another PR later.